### PR TITLE
Create thought spot client in extract

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

If crawler throw exception in the constructor we won't get correct run.metadata.

### 🤓 What?

- Move the connection logic to `extract` for crawlers

### 🧪 Tested?

Test these crawlers locally

- Snowflake
- Snowflake Profile
- BigQuery
- BigQuery Profile
- BigQuery Lineage
- Glue
- ThoughtSpot
- UnityCatalog

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ repos:
         exclude: ^(metaphor/dbt/generated/.+)$
 
   - repo: https://github.com/pycqa/isort
-    rev: 5.11.4
+    rev: 5.12.0
```
Comment (elic-eon) [inline .pre-commit-config.yaml:24]: Fix: https://github.com/pre-commit/pre-commit/issues/1504

Comment (elic-eon) [inline .pre-commit-config.yaml:24]: `https://github.com/home-assistant/core/issues/86892`

File: metaphor/snowflake/profile/extractor.py [hunk 2]
```diff
@@ -124,11 +126,12 @@ def _fetch_tables(self, cursor, database: str) -> Dict[str, DatasetInfo]:
 
     def _fetch_columns_async(
         self,
+        connection: SnowflakeConnection,
         tables: Dict[str, DatasetInfo],
     ) -> None:
 
         # Create a map of full_name => [column_info] from information_schema.columns
-        cursor = self._conn.cursor()
+        cursor = connection.cursor()
```
Comment (mars-lan) [inline metaphor/snowflake/profile/extractor.py:134]: `self._conn` doesn't work?

Comment (elic-eon) [reply metaphor/snowflake/profile/extractor.py:134]: Pass the connection in just for unit test

**Review by mars-lan — APPROVED** [review]

**Pull request merged**